arrow/arrow.py

```diff
@@ -101,7 +102,14 @@ def __init__(self, stoichiometry, rates, random_seed=0):
         '''
 
         self.stoichiometry = stoichiometry
-        self.rates = rates
+        if rates.ndim == 1:
+            rates = [[rate] for rate in rates.astype(np.float64)]
+        self.rates_flat, self.rates_lengths, self.rates_indexes = flat_indexes(rates)
+
+        if forms is not None:
+            self.forms = forms
+        else:
+            self.forms = np.zeros(stoichiometry.shape[0], dtype=np.int64)
```

Comment:
Again, good. Leave all the optional functionality in the python and pass one consistent form to the C. Thanks for this.

Reply:
I think it's a bit more elegant to write code like
```python
if forms is None:
    forms = # some sort of default

self.forms = forms
```
which separates out the input parsing from the attribute assignment.  However this is more my preference than a standard, if I'm not mistaken.  This is a common enough case that you'd think there would be a standard but I've never seen mention of one.